keps/6757-failure-recovery/README.md

```diff
@@ -0,0 +1,307 @@
+Such pods, coloquially called "zombie pods", remain terminating until the node is healed or they are manually removed.
+This is a crucial safety measure, as in that scenario the control plane has no way to confirm that the pods actually stopped
+and released all their resources. It is especially relevant for stateful applications and might result in data corruption.
+For example, the stuck pod might still be writing to a `PersistentVolume` when a replacement is started.
+
+Nevertheless, it became clear that in some cases having a mechanism that would unblock replacement pods from starting would be beneficial.
+In particular, `Job`-based `Workload`s with `podReplacementPolicy: Failed` are unable to make progress when this issue occurs and require manual administration intervention.
+
+### Goals
+
+* Maximize the quota usage for `Workload`s by recovering from a common failure pattern that:
+  * Prevents `Job`s using `podReplacementPolicy: Failed` from making progress.
+* Introduce the concept of failure recovery to the Kueue manager configuration API.
+
+### Non-Goals
+
+* Detecting whether a zombie pod was terminated properly and released all it's resources.
+
+## Proposal
+
+* Introduce a controller that manages moves zombie `Pod`s into the `Failed` phase.
+* Introduce an API for the users to:
+  * Enable the controller.
+  * Configure the grace period between the `Pod`'s deletion time and when the transition to `Failed` happens.
+
+### User Stories (Optional)
+
+#### Story 1
+
+I'm a user running a distributed Machine Learning workload based on `Job`.
+I set `podReplacementPolicy: Failed` to prevent duplicate task registration errors during node drains or preemptions,
+as my framework expects exactly one pod per worker index.
+
+Some nodes in my cluster often go offline for long periods of time, during which my job is unable to make progress -
+replacement pods will not be scheduled as long as the node remains offline. I'd like for my jobs to be automatically unblocked
+as soon as possible, so I can make best use of the available resources (since the training process is synchronized,
+there is no risk of data corruption in my case).
+
+### Risks and Mitigations
+
+#### Replacing Pods Which Are Still Running
+
+Starting a replacement pod without a guarantee that the original one was properly terminated carries an inherent risk
+and might result in inconsistencies, as discussed above.
+If this risk is not communicated to the user properly and the feature is not sufficiently explicit in it's behavior,
+the users might mistake it for a general issue within Kueue/the broader Kubernetes ecosystem.
+
+As this risk stems from the behavior of the `kubelet` and control plane themself, it cannot be fully mitigated without major changes in how Kubernetes operates.
+Instead, it should be adequately documented to prevent users without a compatible use-case from using it.
```

Comment:
Add that as a mitigation we make this behavior opt-in on two levels: admin needs to enable in the configuration, and users need to create Pods with labels matching the configured label selector by the admin.